vscode/src/qirGeneration.ts

```diff
@@ -25,15 +25,28 @@ export async function getQirForActiveWindow(): Promise<string> {
     );
   }
 
+  const configuration = vscode.workspace.getConfiguration("Q#");
   // Check that the current target is base profile, and current doc has no errors.
-  const targetProfile = vscode.workspace
-    .getConfiguration("Q#")
-    .get<string>("targetProfile", "full");
+  const targetProfile = configuration.get<string>("targetProfile", "full");
   if (targetProfile !== "base") {
-    throw new QirGenerationError(
-      "Submitting to Azure is only supported when targeting the QIR base profile. " +
-        "Please update the QIR target via the status bar selector or extension settings."
+    const result = await vscode.window.showWarningMessage(
```

Comment:
I explicitly avoided showing UX in this method as it can be called in multiple ways. For example, this is also wired into the command to "Get QIR for current Q# program", which doesn't submit to Azure, so the error message here doesn't make sense.

I think it would be preferable to return an error from this helper and let the top-level command determine how to handle them and display any UX.

Reply:
(Though interestingly it probably does just display the same "Submitting to Azure..." error today :-/ ).

Reply:
Besides the reference to Azure in the message text, is there anything wrong with prompting the user to set the base profile in the "Get QIR" command? This seems like a desirable side effect since we'd be solving the exact same UX problem in the "Get QIR" scenario.

Reply:
Yeah, I think this can go in as-is for now. Just in general, for library/utility type functions that may get called from different commands, probably best to return an error or exception and let the consuming code decide on the UX experience. I think this is fine for now though.

Reply:
I'm thinking we can refactor out the UX from the "smarts" as we build the integration tests. (Which are happening any day now, I promise...